addons/website_form/static/src/scss/website_form.scss

```diff
@@ -52,3 +52,8 @@
 .form-field input[type=file].form-control {
     height: 100%;
 }
+
+// Without this, datepicker dropdown hides uglily inside the modal body
+.o_website_form_modal .modal-body {
+    overflow: inherit;
```

Comment:
I think you will have to live with the fact the datepicker overflows the way it does right now as we don't make specific fixes for that kind of things, it would have to be done for all modal in odoo, probably in another PR... and I don't think `overflow: inherit` will be a generic solution :wink:

Reply:
However, for this case it's needed. Maybe I could add a specific class for this modal and fix it only here?